Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -314,7 +314,7 @@ def gcc_version_formula(name)
     version = name[GNU_GCC_REGEXP, 1]
     gcc_version_name = "gcc@#{version}"
 
-    gcc = Formulary.factory("gcc")
+    gcc = Formulary.factory(preferred_gcc)
```

Comment:
```suggestion
    gcc = Formulary.factory("gcc")
```
This one should stay as is. It translates `gcc-10` to the formula `gcc`, which is correct on both macOS and Linux.